doc/userguide/devguide/extending/output/eve-filetypes.rst

```diff
@@ -0,0 +1,99 @@
+EVE Filetypes
+#############
+
+Introduction
+************
+
+The Suricata EVE/JSON output supports filetypes to extend how
+EVE records are processed and delivered. Custom filetypes
+provide alternatives to standard file output by implementing a
+file-like interface that Suricata can write to. These filetypes can
+send events to databases, sockets, or other destinations, and can even
+perform custom processing on the output before storing it.
+
+EVE Filetype Life Cycle
+***********************
+
+The life-cycle of an EVE filetype along with the callbacks are
+discussed in ``output-eve.h``:
+
+.. literalinclude:: ../../../../../src/output-eve.h
+   :language: c
+   :start-at: /** \brief Structure used to define an EVE output
```

Comment:
here and in other literalincludes I worry about pattern being too fragile. In some places we use magic comments to mark the start and end of a relevant block. That way it's explicit and no accidental breakage should happen.

Reply:
CI will fail if they don't match. I think we had one case where a struct changed meaning but the magic comments didn't change, so became wrong as well. So I'm not sure if one is better than the other.